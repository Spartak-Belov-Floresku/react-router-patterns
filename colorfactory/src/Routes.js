import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom"

import ColorList  from "./ColorList";
import NewColorForm from "./NewColorForm";
import Color from "./Color";



const Routes = () => {

    return (
        <BrowserRouter>
            <Switch>
                <Route exact path="/colors">
                    <ColorList />
                </Route>
                <Route exact path="/colors/new">
                    <NewColorForm />
                </Route>
                <Route path="/colors/:name">
                    <Color />
                </Route>
                <Redirect to="/colors" />
            </Switch>
        </BrowserRouter>
    );
    
}

export default Routes;