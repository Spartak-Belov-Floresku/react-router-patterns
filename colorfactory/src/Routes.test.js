import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Routes from './Routes';


test('renders learn react link', () => {
    render(<Routes/>);
});

it("matches snapshot", () => {
    const {asFragment} = render(<Routes />);
    expect(asFragment()).toMatchSnapshot();
});

test('renders all colors in the color list', () => {
    const { getByText } = render(
      <MemoryRouter initialEntries={["/colors"]}>
        <Routes />
      </MemoryRouter>
    );
    
    const linkColor = getByText("red");
    expect(linkColor).toBeInTheDocument();

});

