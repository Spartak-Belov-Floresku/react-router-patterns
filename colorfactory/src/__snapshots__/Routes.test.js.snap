// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="App-header"
  >
    <h5>
      Welcom to the color factory.
    </h5>
    <h3>
      <a
        href="/colors/new"
      >
        Add a color
      </a>
    </h3>
  </header>
  <div
    class="ColorList-intro"
  >
    <p>
      Please select a color.
    </p>
    <ul>
      <li>
        <a
          href="/colors/red"
        >
          red
        </a>
      </li>
      <li>
        <a
          href="/colors/green"
        >
          green
        </a>
      </li>
      <li>
        <a
          href="/colors/blue"
        >
          blue
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
